src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, Props, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type searchMethodOptions = "equals" | "lessThan" | "lessThanOrEquals" | "greaterThan" | "greaterThanOrEquals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setsize: number;
+        _setSize: number;
```

Comment:
support only latest version, so only on of the setSize function will be pressent